zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -335,48 +335,32 @@ public void parseProperties(Properties zkProp) throws IOException, ConfigExcepti
                     throw new ConfigException("Unrecognised peertype: " + value);
                 }
             } else if (key.equals("syncEnabled")) {
-                syncEnabled = Boolean.parseBoolean(value);
+                syncEnabled = parseBoolean(key, value);
             } else if (key.equals("dynamicConfigFile")) {
                 dynamicConfigFileStr = value;
             } else if (key.equals("autopurge.snapRetainCount")) {
                 snapRetainCount = Integer.parseInt(value);
             } else if (key.equals("autopurge.purgeInterval")) {
                 purgeInterval = Integer.parseInt(value);
             } else if (key.equals("standaloneEnabled")) {
-                if (value.toLowerCase().equals("true")) {
-                    setStandaloneEnabled(true);
-                } else if (value.toLowerCase().equals("false")) {
-                    setStandaloneEnabled(false);
-                } else {
-                    throw new ConfigException("Invalid option "
-                                              + value
-                                              + " for standalone mode. Choose 'true' or 'false.'");
-                }
+                setStandaloneEnabled(parseBoolean(key, value));
```

Comment:
We could have a version of parseBoolean that reports a better error message.
Otherwise only the stacktrace can give an hint about which parameter is wrong

I suggest to pass the name of the property to parseBoolean

Reply:
Hi @eolivelli , thank you for your review.
I think the current `parseBoolean` already takes the name of the property -- `key`,
and will log out `key` when throwing exception to tell users which parameter is wrong.